sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -8,43 +8,110 @@
 import com.azure.core.annotation.ServiceClient;
 import com.azure.core.annotation.ServiceMethod;
 import com.azure.core.http.rest.Response;
+import com.azure.core.util.Context;
+import com.azure.security.attestation.implementation.AttestationClientImpl;
 import com.azure.security.attestation.implementation.AttestationsImpl;
+import com.azure.security.attestation.implementation.MetadataConfigurationsImpl;
+import com.azure.security.attestation.implementation.SigningCertificatesImpl;
 import com.azure.security.attestation.models.AttestOpenEnclaveRequest;
 import com.azure.security.attestation.models.AttestSgxEnclaveRequest;
 import com.azure.security.attestation.models.AttestationResponse;
+import com.azure.security.attestation.models.AttestationSigner;
 import com.azure.security.attestation.models.CloudErrorException;
 import reactor.core.publisher.Mono;
 
 import java.nio.charset.StandardCharsets;
 import java.util.Objects;
 
+import static com.azure.core.util.FluxUtil.withContext;
+
 /** Initializes a new instance of the asynchronous AzureAttestationRestClient type. */
-@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+@ServiceClient(builder = AttestationAsyncClientBuilder.class, isAsync = true)
 public final class AttestationAsyncClient {
-    private final AttestationsImpl serviceClient;
+    private final AttestationsImpl attestImpl;
+    private final MetadataConfigurationsImpl metadataImpl;
+    private final SigningCertificatesImpl signerImpl;
 
     /**
      * Initializes an instance of Attestations client.
      *
-     * @param serviceClient the service client implementation.
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAsyncClient(AttestationClientImpl clientImpl) {
+        this.attestImpl = clientImpl.getAttestations();
+        this.metadataImpl = clientImpl.getMetadataConfigurations();
+        this.signerImpl = clientImpl.getSigningCertificates();
+    }
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * @param context - Context for the operation
+     * @throws CloudErrorException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return any object.
+     */
+//    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Object>> getOpenIdMetadataWithResponse(Context context) {
```

Comment:
>Object [](http://example.com/codeflow?start=25&length=6)

Do we not want this to be a strongly typed object? More of a design question than implementation.

Reply:
It's weakly typed in the swagger, and in all the other 3 SDKs. In all honesty, I agree that it's a mistake, but fixing it would require creating a new API version for our service, and IMHO this issue doesn't really rise to the level of requiring a new API version.

Reply:
Changing from loosely coupled types to strongly coupled types require a swagger API version change - I did not know this is a breaking change in the swagger eyes - TIL!

Any existing customer code in this scenario won't break - so why is it a breaking change - do you know?

Reply:
Because in strongly typed languages like C# or Java, the machine generated type changes from Object to <something else>, which will generate compilation errors. The swagger folks don't like things which cause existing code to stop compiling (we ran into last month when we wanted to add a new property to one of our existing model types).

Reply:
Makes sense. Thank you :)